langchain/audio_models/bananadev.py

```diff
@@ -0,0 +1,82 @@
+from typing import Dict, Optional
+
+from pydantic import Extra, root_validator
+
+import langchain
+from langchain.audio_models.base import AudioBase
+from langchain.schema import Generation
+from langchain.utils import get_from_dict_or_env
+
+
+class AudioBanana(AudioBase):
+    """Wrapper around Whisper models on Banana.
+
+    To use, you should have the ``banana-dev`` python package installed,
+    and the environment variable ``BANANA_API_KEY`` set with your API key.
+
+    Any parameters that are valid to be passed to the call can be passed
+    in, even if not explicitly saved on this class.
+    """
+
+    model_key: str = ""
+    """model endpoint to use"""
+
+    banana_api_key: Optional[str] = None
+
+    max_chars: Optional[int] = None
+
+    class Config:
+        """Configuration for this pydantic config."""
+
+        extra = Extra.forbid
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        banana_api_key = get_from_dict_or_env(
+            values, "banana_api_key", "BANANA_API_KEY"
+        )
+        values["banana_api_key"] = banana_api_key
+        return values
+
+    def transcript(self, audio_path: str) -> str:
+        """Call to Banana endpoint."""
+        try:
+            import banana_dev as banana
+        except ImportError:
+            raise ValueError(
+                "Could not import banana-dev python package. "
+                "Please install it with `pip install banana-dev`."
+            )
+        api_key = self.banana_api_key
```

Comment:
not sure this or the line below really add anything, dont think super necc

Reply:
we'd also probably want this on the base class right?

imo, let's leave caching logic out for now

Reply:
Agreed.